src/api/entities/Identity/index.ts

```diff
@@ -683,17 +685,22 @@ export class Identity extends Entity<UniqueIdentifiers, string> {
    * Get the list of secondary Accounts related to the Identity
    *
    * @note can be subscribed to
-   * @note This method currently lacks pagination and may be slow for identities with many thousands of keys
+   * @note This method currently can be slow when given a callback or requesting all accounts to be fetched if the Identity has a large number of Accounts
    */
-  public async getSecondaryAccounts(): Promise<PermissionedAccount[]>;
-  public async getSecondaryAccounts(
-    callback: SubCallback<PermissionedAccount[]>
-  ): Promise<UnsubCallback>;
+  public async getSecondaryAccounts(args?: {
+    opts?: PaginationOptions;
+  }): Promise<ResultSet<PermissionedAccount>>;
+
+  public async getSecondaryAccounts(args: {
+    callback: SubCallback<PermissionedAccount[]>;
+  }): Promise<UnsubCallback>;
 
   // eslint-disable-next-line require-jsdoc
-  public async getSecondaryAccounts(
-    callback?: SubCallback<PermissionedAccount[]>
-  ): Promise<PermissionedAccount[] | UnsubCallback> {
+  public async getSecondaryAccounts(args: {
+    callback?: SubCallback<PermissionedAccount[]>;
+    opts?: PaginationOptions;
+    fetchAll?: boolean;
+  }): Promise<ResultSet<PermissionedAccount> | UnsubCallback> {
```

Comment:
For consistency, `callback` should be a separate parameter. Use `AccountManagement.getAccountBalance` for reference. Also, I assume pagination options and `fetchAll` are not meant to be passed together, so there should be two separate signatures where each of them is defined and the other isn't. This means a lot of possible combinations when mixed with the optional callback, so I suggest defining a separate type for the first argument

Also, you can type `fetchAll` as `true` (you should also document it or at least use the exact property name in the note)